src/main/java/duke/Storage.java

```
@@ -0,0 +1,82 @@
+                bufferedWriter.write(t.toSaveString());
+                bufferedWriter.newLine();
+            }
+            bufferedWriter.flush();
+            bufferedWriter.close();
+        } catch (IOException e) {
+            Ui.Error(e);
+        }
+    }
+
+    public ArrayList<Task> loadSaveFile() {
+        FileReader fileReader;
+        BufferedReader bufferedReader;
+        ArrayList<Task> input = new ArrayList<Task>();
+        try {
+            File outputFile = new File(dir);
+            outputFile.getParentFile().mkdirs();
+            if (!outputFile.exists()) {
+                outputFile.createNewFile();
+            }
+            fileReader = new FileReader(dir);
+            bufferedReader = new BufferedReader((fileReader));
+            String nextLine;
+            nextLine = bufferedReader.readLine();
+            while (nextLine != null) {
+                String[] splitted = nextLine.split(" \\| ");
+                switch (splitted[0]) {
+                case "T":
+                    input.add(new ToDo(splitted[2], (splitted[1] == "1")));
+                    break;
+                case "E":
+                    input.add(new Event(splitted[2], splitted[3], splitted[4], (splitted[1] == "1")));
+                    break;
+                case "D":
+                    int key = Integer.valueOf(splitted[4]);
+                    if (key == 2) {
+                        input.add(new Deadline(splitted[2], LocalDateTime.parse(splitted[3]), (splitted[1] == "1")));
+                    } else if (key == 1) {
+                        input.add(new Deadline(splitted[2], LocalDate.parse(splitted[3]), (splitted[1] == "1")));
+                    } else {
+                        Ui.badDateLoaded();
+                    }
+                    break;
+                }
+                nextLine = bufferedReader.readLine();
+            }
+            bufferedReader.close();
+        } catch (IOException e) {
```

i think it's better to throw different types of error instead of printing the error messages